ctapipe/io/eventfilereader.py

```diff
@@ -2,38 +2,75 @@
+
+    A new EventFileReader should be created for each type of event file read
+    into ctapipe, e.g. sim_telarray files are read by the `HessioFileReader`.
+
+    EventFileReader provides a common high-level interface for accessing event
+    information from different data sources (simulation or different camera
+    file formats). Creating an EventFileReader for a new
+    file format ensures that data can be accessed in a common way,
+    irregardless of the file format.
+
+    EventFileReader itself is an abstract class. To use an EventFileReader you
+    must use a subclass that is relevant for the file format you
+    are reading (for example you must use
+    `ctapipe.io.hessiofilereader.HessioFileReader` to read a hessio format
+    file). Alternatively you can use
+    `ctapipe.io.eventfilereader.EventFileReaderFactory` to automatically
+    select the correct EventFileReader subclass for the file format you wish
+    to read.
+
+    To create an instance of an EventFileReader you must pass the traitlet
+    configuration (containing the input_path) and the
+    `ctapipe.core.tool.Tool`. Therefore from inside a Tool you would do:
+
+    >>> reader = EventFileReader(self.config, self)
+
+    An example of how to use `ctapipe.core.tool.Tool` and
+    `ctapipe.io.eventfilereader.EventFileReaderFactory` can be found in
+    ctapipe/examples/calibration_pipeline.py.
+
+    However if you are not inside a Tool, you can still create an instance and
+    supply an input_path via:
+
+    >>> reader = EventFileReader(None, None, input_path="/path/to/file")
+
+    To loop through the events in a file:
+
+    >>> reader = EventFileReader(None, None, input_path="/path/to/file")
+    >>> for event in reader:
+    >>>    print(event.count)
+
+    **NOTE**: Every time a new loop is started through the reader, it restarts
+    from the first event.
+
+    Alternatively one can use EventFileReader in a `with` statement to ensure
+    the correct cleanups are performed when you are finished with the reader:
+
+    >>> with EventFileReader(None, None, input_path="/path/to/file") as reader:
+    >>>    for event in reader:
+    >>>       print(event.count)
 
     Attributes
     ----------
     input_path : str
-    directory : str
-        Automatically set from `input_path`.
-    filename : str
-        Name of the file without the extension.
-        Automatically set from `input_path`.
-    extension : str
-        Automatically set from `input_path`.
-    output_directory : str
-        Directory to save outputs for this file
-
+        Path to the input event file.
+    max_events : int
+        Maximum number of events to loop through in generator
     """
-    origin = None
 
-    input_path = Unicode(get_dataset('gamma_test.simtel.gz'), allow_none=True,
+    input_path = Unicode('', allow_none=False,
                          help='Path to the input file containing '
                               'events.').tag(config=True)
```

Comment:
Just a side note:
`allow_none=False` is the default of `Unicode`. 

Indentation: I think this way of indenting long lines is sub-optimal
```python
input_path = Unicode('', allow_none=False,
                     help='Path to the input file containing '
                     'events.').tag(config=True)
```

Indenting like this:
```python
input_path = Unicode(
    '', 
    allow_none=False,
    help='Path to the input file containing events.'
).tag(config=True)
```

has the following advantages:
 * No need to modify indentation, when `Unicode` gets replaced by a different class  (i.e. stable under refactoring / renaming)
 * generally more space available for parameters, in this case, the help string does not need to be broken around.
 * in case of testing: single parameters can be commented out more easily, since each is on their own line.
```